storage/local/persistence.go

```diff
@@ -247,11 +252,29 @@ func newPersistence(
 				Help:      "Quantiles for batch indexing duration in seconds.",
 			},
 		),
-		checkpointDuration: prometheus.NewGauge(prometheus.GaugeOpts{
+		checkpointLastDuration: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "checkpoint_last_duration_seconds",
+			Help:      "The duration in seconds it took to last checkpoint in-memory metrics and head chunks.",
```

Comment:
The "head chunks" part is kind of obsolete. How about "... and non-persisted chunks"?

In different news, the verbing of the noun "checkpoint" reads a bit weird, in particular in more complex syntax like with "last". How about:
```
"The duration it took to create the last checkpoint of in-memory metrics and non-persistent chunks."
```